functions-framework-api/src/main/java/com/google/cloud/functions/ExperimentalCloudEventsFunction.java

```diff
@@ -0,0 +1,22 @@
+package com.google.cloud.functions;
+
+import io.cloudevents.CloudEvent;
+
+/**
+ * Represents a Cloud Function that is activated by an event and parsed into a {@link CloudEvent} object.
+ * Because the {@link CloudEvent} API is not yet stable, a function implemented using this class may not
+ * build or work correctly with later versions of that API. Once the API is stable, this interface will
+ * become {@code CloudEventsFunction} and will also be stable.
+ */
+@FunctionalInterface
+public interface ExperimentalCloudEventsFunction {
+  /**
+   * Called to service an incoming event. This interface is implemented by user code to
+   * provide the action for a given background function. If this method throws any exception
+   * (including any {@link Error}) then the HTTP response will have a 500 status code.
+   *
+   * @param event the event.
+   * @throws Exception to produce a 500 status code in the HTTP response.
+   */
+  void accept(CloudEvent event) throws Exception;
```

Comment:
So this is what I refer to as an "untyped" CloudEvent function in C#.
Is there an expectation that you'd provide a "typed" CloudEvent function as well, where the author specifies the expected data type and you do the unmarshalling for them, or is the expectation that it's sufficiently easy to unmarshal the data, so leave it to the user?

Reply:
I've gone back and forth a bit on this. For legacy events, we already have `RawBackgroundFunction` and `BackgroundFunction`, which are respectively untyped and typed. We could do the same for CloudEvents. As you know, there's also discussion in the Java SDK about whether it should provide support for this kind of automatic unmarshalling, which in fact the v1 SDK did. I think it's probably not the SDK's place to do that.

Assuming the `byte[] getData()` of the payload is a JSON string, users can do this themselves with something like this...

```
PubSub message = new Gson().fromJson(new String(event.getData(), UTF_8), PubSub.class);
```

...which is maybe just annoying enough for us to provide a way to have it done for them. The disadvantage of that is that we have to choose a particular JSON framework, which might not be the same as the one they're using.

Since the API here is necessarily unstable (because the SDK is), we could in fact postpone making this particular decision until finalizing it. We can still go ahead with implementing and testing CloudEvents support, which is the bulk of the work.